dashboard/client/src/pages/serve/ServeDeploymentsListPage.tsx

```diff
@@ -89,82 +88,36 @@ export const ServeDeploymentsListPage = () => {
-                      "name",
-                      value.trim() !== "-" ? value.trim() : "",
-                    );
-                  }}
-                  renderInput={(params: TextFieldProps) => (
-                    <TextField {...params} label="Name" />
-                  )}
-                />
-                <Autocomplete
-                  style={{ margin: 8, width: 120 }}
-                  options={Array.from(
-                    new Set(allServeDeployments.map((e) => e.status)),
-                  )}
-                  onInputChange={(_: any, value: string) => {
-                    changeFilter("status", value.trim());
-                  }}
-                  renderInput={(params: TextFieldProps) => (
-                    <TextField {...params} label="Status" />
-                  )}
-                />
-                <Autocomplete
-                  style={{ margin: 8, width: 120 }}
-                  options={Array.from(
-                    new Set(allServeDeployments.map((e) => e.applicationName)),
-                  )}
-                  onInputChange={(_: any, value: string) => {
-                    changeFilter("applicationName", value.trim());
-                  }}
-                  renderInput={(params: TextFieldProps) => (
-                    <TextField {...params} label="Application" />
-                  )}
-                />
-                <TextField
-                  style={{ margin: 8, width: 120 }}
-                  label="Page Size"
-                  size="small"
-                  defaultValue={10}
-                  InputProps={{
-                    onChange: ({ target: { value } }) => {
-                      setPage("pageSize", Math.min(Number(value), 500) || 10);
-                    },
-                    endAdornment: (
-                      <InputAdornment position="end">Per Page</InputAdornment>
-                    ),
-                  }}
-                />
-              </div>
-              <div style={{ display: "flex", alignItems: "center" }}>
-                <Pagination
-                  count={Math.ceil(
-                    filteredServeDeployments.length / page.pageSize,
-                  )}
-                  page={page.pageNo}
-                  onChange={(e, pageNo) => setPage("pageNo", pageNo)}
-                />
-              </div>
+              <TextField
+                style={{ margin: 8, width: 120 }}
+                label="Page Size"
+                size="small"
+                defaultValue={10}
+                InputProps={{
+                  onChange: ({ target: { value } }) => {
+                    setPage("pageSize", Math.min(Number(value), 500) || 10);
+                  },
+                  endAdornment: (
+                    <InputAdornment position="end">Per Page</InputAdornment>
+                  ),
+                }}
+              />
+              <Pagination
+                count={Math.ceil(serveDeployments.length / page.pageSize)}
```

Comment:
The pagination logic needs to be fixed before we merge. Right now it's calculating the count based on the number of deployments, but below we're using it to slice the Applications array.

One option is to get rid of the page component entirely. Another option is to somehow incorporate the total number of deployments and application rows into the page calculation. The thing to be careful about is we can't split a single application and its dropdown deployments across multiple pages.

Apart from that, this is ready for review @alanwguo @scottsun94. Feel free to suggest other reviewers as well

Reply:
+1 to paginate based on application for now

Reply:
Sounds good. Fixed 4a7d868b1d